vscode-istio.yaml

```diff
@@ -0,0 +1,85 @@
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: code-server-gateway
+spec:
+  selector:
+    istio: ingressgateway # use istio default controller
+  servers:
+  - port:
+      number: 80
+      name: http
+      protocol: HTTP
+    hosts:
+    - "*"
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: code-server
+spec:
+  hosts:
+  - "*"
+  gateways:
+  - code-server-gateway
+  http:
+  - match:
```

Comment:
In my solution, I had multiple match rules.   This solution only has one.  I recall testing this with a different root path string and the ui either did not render or there were missing ui icons.

The blog posting you passed along showed multiple match rules.  That work and AFACIT, no ui issue.  

However, in thinking about it more, the example had a concept of a default route.  I think we can probably get by with a simpler match set by taking advantage of the default route.  This may put it closer to your proposed solution.